syntax = "proto3";

package Actions;

message FileTransactionSetup
{
    string id = 1;
    string hash = 2;
    string topic = 3;
}

message FileTransactionStatus
{
    string fileName = 1;
    string hash = 2;
    string destinationPath = 3;
    uint64 totalChunks = 4;
}

message FileTransactionChunk
{
    string data = 1;
    string hash = 2;
    uint64 chunkNumber = 3;
    uint64 totalChunks = 4;
}